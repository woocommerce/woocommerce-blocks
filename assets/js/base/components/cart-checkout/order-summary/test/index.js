/**
 * External dependencies
 */
import TestRenderer from 'react-test-renderer';
import { __experimentalRegisterCheckoutFilters } from '@woocommerce/blocks-checkout';

/**
 * Internal dependencies
 */
import OrderSummaryItem from '../order-summary-item';

describe( 'OrderSummaryItem', () => {
	test( "should not render a suffix if the filter to change it isn't registered", () => {
		const component = TestRenderer.create(
			<OrderSummaryItem
				cartItem={ {
					quantity: 1,
					images: [],
					name: 'Test item',
					permalink: '/test-item',
					prices: {
						price: '5000',
						regular_price: '61400',
						raw_prices: {
							precision: 6,
							price: '50000000',
							regular_price: '614000000',
							sale_price: '50000000',
						},
					},
					summary: 'Great item',
				} }
			/>
		);

		expect( component.toJSON() ).toMatchSnapshot();
	} );
	test( 'should render a suffix if one is added by a filter', () => {
		__experimentalRegisterCheckoutFilters( 'woocommerce-subscriptions', {
			subtotalSuffix: () => {
				return 'test suffix';
			},
		} );

		const component = TestRenderer.create(
			<OrderSummaryItem
				cartItem={ {
					quantity: 1,
					images: [],
					name: 'Test item',
					permalink: '/test-item',
					prices: {
						price: '5000',
						regular_price: '61400',
						raw_prices: {
							precision: 6,
							price: '50000000',
							regular_price: '614000000',
							sale_price: '50000000',
						},
					},
					summary: 'Great item',
				} }
			/>
		);

		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
