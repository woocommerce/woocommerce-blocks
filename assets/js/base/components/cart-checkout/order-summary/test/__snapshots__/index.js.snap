// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSummaryItem should not render a suffix if the filter to change it isn't registered 1`] = `
<div
  className="wc-block-components-order-summary-item"
>
  <div
    className="wc-block-components-order-summary-item__image"
  >
    <div
      className="wc-block-components-order-summary-item__quantity"
    >
      <span
        aria-hidden="true"
      >
        1
      </span>
      <span
        className="screen-reader-text"
      >
        1 items
      </span>
    </div>
    <img
      alt=""
      src=""
    />
  </div>
  <div
    className="wc-block-components-order-summary-item__description"
  >
    <span
      className="wc-block-components-product-name"
    >
      Test item
    </span>
    <span
      className="wc-block-components-order-summary-item__individual-prices price wc-block-components-product-price"
    >
      <span
        className="screen-reader-text"
      >
        Previous price:
      </span>
      <del
        className="wc-block-components-product-price__regular wc-block-components-order-summary-item__regular-individual-price"
      >
        &#36;614.00
      </del>
      <span
        className="screen-reader-text"
      >
        Discounted price:
      </span>
      <ins
        className="wc-block-components-product-price__value is-discounted wc-block-components-order-summary-item__individual-price"
      >
        &#36;50.00
      </ins>
      
      
    </span>
    <div
      className="wc-block-components-product-metadata"
    />
  </div>
  <div
    className="wc-block-components-order-summary-item__total-price"
  >
    <span
      className="price wc-block-components-product-price"
    >
      <span
        className="wc-block-formatted-money-amount wc-block-components-formatted-money-amount wc-block-components-product-price__value"
      >
        &#36;50.00
      </span>
      
      
    </span>
  </div>
</div>
`;

exports[`OrderSummaryItem should render a suffix if one is added by a filter 1`] = `
<div
  className="wc-block-components-order-summary-item"
>
  <div
    className="wc-block-components-order-summary-item__image"
  >
    <div
      className="wc-block-components-order-summary-item__quantity"
    >
      <span
        aria-hidden="true"
      >
        1
      </span>
      <span
        className="screen-reader-text"
      >
        1 items
      </span>
    </div>
    <img
      alt=""
      src=""
    />
  </div>
  <div
    className="wc-block-components-order-summary-item__description"
  >
    <span
      className="wc-block-components-product-name"
    >
      Test item
    </span>
    <span
      className="wc-block-components-order-summary-item__individual-prices price wc-block-components-product-price"
    >
      <span
        className="screen-reader-text"
      >
        Previous price:
      </span>
      <del
        className="wc-block-components-product-price__regular wc-block-components-order-summary-item__regular-individual-price"
      >
        &#36;614.00
      </del>
      <span
        className="screen-reader-text"
      >
        Discounted price:
      </span>
      <ins
        className="wc-block-components-product-price__value is-discounted wc-block-components-order-summary-item__individual-price"
      >
        &#36;50.00
      </ins>
       
      test suffix
    </span>
    <div
      className="wc-block-components-product-metadata"
    />
  </div>
  <div
    className="wc-block-components-order-summary-item__total-price"
  >
    <span
      className="price wc-block-components-product-price"
    >
      <span
        className="wc-block-formatted-money-amount wc-block-components-formatted-money-amount wc-block-components-product-price__value"
      >
        &#36;50.00
      </span>
      
      
    </span>
  </div>
</div>
`;
